src/block/Composter.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\data\runtime\RuntimeDataDescriber;
+use pocketmine\event\block\ComposterEmptyEvent;
+use pocketmine\event\block\ComposterFillEvent;
+use pocketmine\event\block\ComposterReadyEvent;
+use pocketmine\item\Item;
+use pocketmine\item\VanillaItems;
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+use pocketmine\world\particle\CropGrowthEmitterParticle;
+use pocketmine\world\sound\ComposterEmptySound;
+use pocketmine\world\sound\ComposterFillSound;
+use pocketmine\world\sound\ComposterFillSuccessSound;
+use pocketmine\world\sound\ComposterReadySound;
+
+class Composter extends Transparent{
+	public const MIN_COMPOST_LAYERS = 0;
+	public const MAX_COMPOST_LAYERS = 8;
+
+	protected int $layers = 0;
+
+	protected function describeBlockOnlyState(RuntimeDataDescriber $w) : void{
+		$w->boundedIntAuto(0, self::MAX_COMPOST_LAYERS, $this->layers);
+	}
+	
+	protected function recalculateCollisionBoxes() : array{
+		foreach(Facing::HORIZONTAL as $f){
+			$result[] = AxisAlignedBB::one()->trim($f, 14 / 16);
+		}
+		return $result;
```

Comment:
Shouldn't this also have an AABB for the bottom of the composter, so entities can't fall through it?

Reply:
This also needs to consider the number of layers